File: sga/activation.py
import math


def identity(x):
    return x


def heaviside(x):
    return 1 if x >= 0 else 0


def sigmoid(x):
    return 1 / (1 + math.exp(x))


def tanh(x):
    return math.tanh(x)


def atan(x):
    return math.atan(x)


def relu(x):
    return max(x, 0)


def softplus(x):
    return math.log(1 + math.exp(x), math.e)


def sin(x):
    return math.sin(x)


def gauss(x):
    return math.exp(-x**2)

File: src/main/java/com/github/gmattis/simplegeneticalgorithm/util/Activation.java
package com.github.gmattis.simplegeneticalgorithm.util;

public enum Activation {
	
	IDENTITY("identity"),
	HEAVISIDE("heaviside"),
	SIGMOID("sigmoid"),
	RELU("relu"),
	SOFTPLUS("softplus"),
	GAUSS("gauss");
	
	private String name;

	Activation(String name) {
		this.name = name;
	}
	
	public String getName() {
		return name;
	}
	
	//////////////////////////
	// Activation Functions //
	//////////////////////////

	public static Double identity(Double x) {
		return x;
	}
	
	public static Double heaviside(Double x) {
		if (x >= 0) {
			return 1.;
		} else {
			return 0.;
		}
	}
	
	public static Double sigmoid(Double x) {
		return 1. / (1. + Math.exp(x));
	}
	
	public static Double relu(Double x) {
		return Math.max(0., x);
	}
	
	public static Double softplus(Double x) {
		return Math.log(1. + Math.exp(x));
	}
	
	public static Double gauss(Double x) {
		return Math.exp(-(x*x));
	}
}
